### PR TITLE
support nkey file as string

### PR DESCRIPTION
**Pull request by gangtao**
- adding a new fields `nkey_file_content`
- create a tmp file before connection
- delete this tmp file after connection complete

### CODE REVIEW

File: internal/impl/nats/auth.go
Comment (zliang-min) [inline internal/impl/nats/auth.go]: Do not hard code the tmp dir and file name (multiple sources/sinks could get conflict). Use `os.MkdirTemp` instead (it will use the temp dir from system, so it makes sure you code works on all supported OS's, not just Linux):

```go
	dir, err := os.MkdirTemp("", "nats")
	if err != nil {
		// handle err
	}
        nkeyFile := filepath.Join(dir, "nkeys")
        // create the file and write content to the file ...
```

**Review by zliang-min — CHANGES_REQUESTED** [review]
One small change.

File: internal/impl/nats/auth/docs.go
Comment (ye11ow) [inline internal/impl/nats/auth/docs.go]: `An optional file` - you don't pass the file to this option right? `An optional NKey seed string` or something like that is more accurate

File: internal/impl/nats/input.go
Comment (ye11ow) [inline internal/impl/nats/input.go]: hmm, is this file safe to delete at this moment? Will the lib reads it later?

Comment (gangtao) [reply internal/impl/nats/input.go]: I tested it, only read at connection

File: internal/impl/nats/auth.go
Comment (gangtao) [reply internal/impl/nats/auth.go]: is that a random path?  I need know this path later to delete it. otherwise there will be many temp file left

Comment (gangtao) [reply internal/impl/nats/auth.go]: we can work around by just delete all temp file for nkeys, there is a rare chance it deleted other nats's nkey file

Comment (gangtao) [reply internal/impl/nats/auth.go]: I am wipe it in another routin after wait 60s, which seem works

File: internal/impl/nats/auth.go
Comment (zliang-min) [inline internal/impl/nats/auth.go]: I checked the nats client code, the options actually is stored in the client, and there is logic in the client that if failure happens, it could reconnect with the same options. So, I think the best way to handle the deletion is to delte the file when the `Close` function on the component is called.

Comment (gangtao) [reply internal/impl/nats/auth.go]: the problem is that we cannot transfer this file path information from auth to nats input, or we have to keep that tmp file

File: internal/impl/nats/auth.go
Comment (zliang-min) [inline internal/impl/nats/auth.go]: Delete the unused variable. If we enable lint checks, this will fail.

**Review by zliang-min — APPROVED** [review]
Still one small issue need to be fixed.

Approved the PR so that once it's fixed, we can merge it without another reivew.

**Comment by jovezhong** [comment]
(Jove Github Bot) added it to the current sprint.

**Comment by jovezhong** [comment]
(Jove Github Bot) moved this ticket out of the GitHub project(up to 1200 tickets for one project).

**Pull request merged**
